Failure to decode cut-off UTF-8

I can't post the original example because the sources involved are not open source but I suspect the problem is easy enough to find from this part here:
```
    statement = b'ome x.\n      Axiom to_of_N : \xe2\x88\x80 n x, of_N n = Some x \xe2\x86'
    statement_str = ' x) = S'
    ty = 'lib'
    types = ('lib',)
UnicodeDecodeError: 'utf-8' codec can't decode bytes in position 54-55: unexpected end of data
```

The code around this excerpt looks like this:
```coq
      Axiom of_to_N : ∀ x, of_N (to_N x) = Some x.
      Axiom to_of_N : ∀ n x, of_N n = Some x → to_N x = n.
```
The UTF-8 representation of `→` is `0xE2 0x86 0x92`.

I suspect something is splitting raw bytes instead of unicode code points. 

I cannot easily reproduce the problem with smaller files so there might be some buffering/chunking going on? Maybe a chunk's border lands between the three bytes required for `→`?

I checked the output again and I now see this here which might explain why the `.glob` file is out of date:
```
make -k -f Makefilessw5xofk.coq KEEP_ERROR=1 fmdeps/cpp2v/tests/bug5_minimal.glob
make: /bin/sh: Argument list too long
COQDEP VFILES
make: /bin/sh: Argument list too long
make: *** No rule to make target 'fmdeps/cpp2v/tests/bug5_minimal.glob'.
```

--- a/coq_tools/find_bug.py
+++ b/coq_tools/find_bug.py
@@ -1018,8 +1018,12 @@ def try_split_requires(output_file_name, **kwargs):
         kwargs['log']('\nNon-fatal error: Failed to get references for %s' % output_file_name, level=LOG_ALWAYS)
         return False
 
-    annotated_contents = split_requires_of_statements(annotated_contents, **kwargs)
-    new_contents = ''.join(v[0].decode('utf-8') for v in annotated_contents)
+    try:
+        annotated_contents = split_requires_of_statements(annotated_contents, **kwargs)
+        new_contents = ''.join(v[0].decode('utf-8') for v in annotated_contents)
+    except UnicodeDecodeError as e:
+        kwargs['log']('\nNon-fatal error: Invalid non-UTF-8 (maybe an outdated .glob file for %s?): %s' % (output_file_name, str(e)), level=LOG_ALWAYS)
+        return False
 
     return check_change_and_write_to_file(old_contents, new_contents, output_file_name,
                                           unchanged_message='No Requires to split.',

